### PR TITLE
Add multiple selection with marquee

### LINKED ISSUE #820: As a xoder I want to select multiple items with marquee because that’s intuitive
### Rationale

We have multiple selection, but not a rectangular marquee. The UX will improve much if there will be such feature. It would let making patches faster.

The question which often arises is “Should I include items only they’re completely inside the rectangle or touching it is enough”. The problem solved wisely in QCad. If you grow the marquee left-to-right only fully covered items are included, and when you grow right-to-left even touched items are included.

qcad-marquee

Note how a border of the rectangle differs in two modes.

### Acceptance criteria

- [x] I can select multiple items with marquee with mechanics described
- [x] I can hold Ctrl at the moment of mouse button release and the selection would be added to the existing

### PR DESCRIPTION
**Pull request by brusherru**
It closes #820.

Preview
On preview:
Black circle — just left click
Red circle — left click + cmd (or ctrl)

### CODE REVIEW

File: packages/xod-client/src/editor/containers/Patch/modes/marqueeSelecting.jsx [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+      isMouseDown: true,
+      dragStartPosition,
+      mousePosition,
+      selection: props.selection,
+    };
+  },
+  onMouseMove(api, event) {
+    api.setState({
+      mousePosition: getMousePosition(patchSvgRef, api.props.offset, event),
+      selection: getComputedSelection(api, event),
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/containers/Patch/modes/marqueeSelecting.jsx]: Looks like selection is calculated with previous mouse position

File: packages/xod-client/src/editor/containers/Patch/modes/marqueeSelecting.jsx [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+      <rect
+        className={cls}
+        x={from.x}
+        y={from.y}
+        width={width}
+        height={height}
+      />
+    );
+  },
+  renderBlockingLayer(api) {
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/containers/Patch/modes/marqueeSelecting.jsx]: A comment about what exactly this layer blocks(patch entities from mouse events?) would be nice

File: packages/xod-client/src/editor/containers/Patch/modes/selecting.jsx [hunk 3]
```diff
@@ -39,25 +40,31 @@ const selectingMode = {
...
       props.actions.addEntityToSelection(entityType, entityId);
       setState({
+        isMouseDown: true,
         isMouseDownOnMovableObject: true,
         dragStartPosition: mousePosition,
       });
     } else {
       props.actions.selectEntity(entityType, entityId);
       setState({
+        isMouseDown: true,
```
Comment (evgenykochetkov) [inline packages/xod-client/src/editor/containers/Patch/modes/selecting.jsx]: I don't think all those `isMouseDown`s are necessary. Maybe, a single `isMouseDownOnBackground` is clearer?

**Comment by brusherru** [comment]
@evgenykochetkov @nkrkv check it out, please :)

**Pull request merged**